File: pickerkit/src/main/java/com/dewarder/pickerkit/utils/Recyclers.java
package com.dewarder.pickerkit.utils;

import android.support.v7.widget.RecyclerView;

public final class Recyclers {

    private Recyclers() {
        throw new UnsupportedOperationException();
    }

    public static int calculateSpanCount(RecyclerView recyclerView, int minItemSize) {
        return recyclerView.getWidth() / minItemSize;
    }

    public static int calculateItemSize(RecyclerView recyclerView, int spanCount, int minItemSize, int itemSpacing) {
        int recyclerWidth = recyclerView.getWidth();
        return (int) (minItemSize + (float) (recyclerWidth - minItemSize * spanCount - itemSpacing * (spanCount + 1)) / spanCount);
    }
}

File: pickerkit/src/main/java/com/dewarder/pickerkit/utils/Recyclers.kt
package com.dewarder.pickerkit.utils

import android.support.v7.widget.RecyclerView

object Recyclers {

    fun calculateSpanCount(recyclerView: RecyclerView, minItemSize: Int): Int {
        return recyclerView.width / minItemSize
    }

    fun calculateItemSize(
            recyclerView: RecyclerView,
            spanCount: Int,
            minItemSize: Int,
            itemSpacing: Int
    ): Int {
        val recyclerWidth = recyclerView.width
        return (minItemSize + (recyclerWidth - minItemSize * spanCount - itemSpacing * (spanCount + 1)).toFloat() / spanCount).toInt()
    }
}
